### ISSUE TITLE
Private geth blockchain not connecting

### ISSUE THREAD

**Issue by BlockChange** [ethereum/go-ethereum#2761]
I have some things working with `testrpc` and `web3` and `ether-pudding` etc.

Now I'd like to create some sort of staging environment which has 2 or more Nodes running, with `geth`

Now I've tried this: 

`geth --identity "MyNodeName" --genesis CustomGenesis.json --rpc --rpcport "8080" --rpccorsdomain "*" --datadir "~/blockchains/VPSChain" --port "30303" --nodiscover --ipcapi "admin,db,eth,debug,miner,net,shh,txpool,personal,web3" --rpcapi "db,eth,net,web3" --autodag --networkid 1900 --nat "any" --dev console`

CustomGenesis.json:
```
{
    "nonce": "0x0000000000000042",
    "timestamp": "0x0",
    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "extraData": "0x0",
    "gasLimit": "0x8000000",
    "difficulty": "0x400",
    "mixhash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "coinbase": "0x3333333333333333333333333333333333333333",
    "alloc": {
    }
}
```

This runs, but when I try to connect to port `8080` it is not serving me anything, I expected this to give me some kind of status, or like the `testrpc` does: `400 bad request`. But it serves nothing as far as I can see.
Also I don't know what to do with port `30303`, I tried connecting to it with `web3` and also to `8080`, but it says `ERR_CONNECTION_REFUSED`.

So in short:
What should be configured to go from `testrpc` to `geth`?

**Comment by nimmaj** [comment]
fwiw we use a similar custom genesis and this (in docker, but you can just run it locally):

ENTRYPOINT ["./go-ethereum/build/bin/geth"]

CMD ["--networkid", "234567", "--rpc", "--rpcaddr", "0.0.0.0", "--nodiscover", "--rpcapi", "eth,net,web3,admin,miner,personal"]

these days we also:

./go-ethereum/build/bin/geth init ./CustomGenesis.json 

before running.  

We're connecting on port 8545.  

Worth noting that our system (which also uses ether pudding) stopped working properly sometime between geth 1.4.3 and 1.4.7 - still trying to figure out why.  We're running a miner and a bunch of non-mining geth clients - you can send txns from the miner to the clients, but the other way around they just get stuck.  1.4.3 is fine.

Ping me if this doesn't help.  I can always send you our docker file.

**Comment by wawrzek** [comment]
I also notice private networks stopped to work properly recently.

**Comment by brendan87** [comment]
I'm trying to set up a private geth single node chain and I am encountering the following:
Using latest geth (1.4.9) and connecting via [browser-solidity](http://ethereum.github.io/browser-solidity/#version=soljson-latest.js), mined contracts don't appear to be running their constructors, since the 'owner' address is set to '0x0' rather than 'msg.sender'.
Using geth 1.4.3 as suggested above results in the solidity browser interface reporting that no code exists at addresses after mining has been done.

**Comment by wawrzek** [comment]
@karalabe Could you add 'Private Network' label please.

**Comment by karalabe** [comment]
The RPC endpoint (8545 by default) speaks the JSON RPC protocol. If you just connect to it without sending anything and wait it doesn't know what to do as it itself is waiting for a request to reply to. If you would execute an HTTP `GET` against it (e.g. from a browser window), you get back `{"jsonrpc":"2.0","error":{"code":-32600,"message":"EOF"}}` so that seems fine to me.

Port 30303 is the devp2p port that we use for P2P data exchange, to run the Ethereum network. You shouldn't attach to that.

With regard to transaction propagation, there was an update to fix a sync issue that didin't consider a single miner scenario. It's a pending bug tracked in https://github.com/ethereum/go-ethereum/issues/2769.
